### DISCUSSION TITLE
posts/install-zerotier-on-steamdeck/

### DISCUSSION THREAD

**Discussion by giscus** [flexiondotorg/flexiondotorg.github.io#197] [Blog Comments]
# posts/install-zerotier-on-steamdeck/

Blog, Podcasts, Live Streams and Videos about Linux and Open Source with a focus on Ubuntu, Debian and NixOS.

https://wimpysworld.com/posts/install-zerotier-on-steamdeck/

**Comment by Mattj172** [comment]
Cheching package integrity error libnatpmp: signature from gitlab Ci package builder is unkown trust

**Reply by gaodi16366** [reply]
same error, changed etc/pacman.config to "SigLevel = Never" but not work

**Reply by gaodi16366** [reply]
solve this by enter "n" when asked if delete file, then fix etc/pacman.config to "LocalSigLevel = Never" and install download zst file by "pacman -U --noconfirm /var/cache/pacman/pkg/xxxxxx.pkg.tar.zst "

**Comment by Leorscaldas** [comment]
Hi I was able to install the program with sucess, yet I`m unable to locate the application to join other networks. And when a I look in Konsole I` unable to find and execute the application

Can you help me, please?

**Reply by aaron777collins** [reply]
You'd need to use the zerotier-cli to join networks (sudo zerotier-cli join <network-id>). If you want a GUI application, check out my [reddit post](https://www.reddit.com/r/SteamDeck/comments/1dllbbl/install_zerotierone_static_a_frontend_gui_flatpak/). I made an installer that packages a static binary backend and a flatpak front-end. It also works in game-mode. However, if you used this method to install zero-tier, you may want to uninstall it before using my installer. It will likely overlap. Optionally, you can disable it by running `sudo systemctl disable zerotier-one`. Let me know how it goes. Thanks!

**Comment by correii** [comment]
# Hey dude, this bricked my Steam Deck. Like to the point where I can't even get it to boot from a recovery image.

I entered the commands exactly as you said. The problem is that you failed to include the warning that's on the Rwfus GitHub page saying it can brick your Steam Deck. Which would've been a nice warning to have before I followed your instructions and lost literally everything on my SSD. It looks like I'm gonna have to send it back to Valve to get it working again, since like I said, it won't even boot from either of the two USB sticks I flashed with the SteamOS recovery image. 

I can not overstate how devastated I am by this. Please edit this guide to include the warning, or at least delete it so this doesn't happen to anyone else. I have a lot of choice words that I'm trying very hard to keep to myself right now in hopes that you will actually fix/delete the guide. If you don't, **this _will_ inevitably happen to other people**. I'm sure you don't want that.

Here is a link to the warning to make it easy for you: https://github.com/ValShaped/rwfus?tab=readme-ov-file#-jank-warning

### Now I need to sign off before I scream at you.

**Reply by gbdrbob** [reply]
Dude,  I don’t know what you did but the jank/brick warning is about not running a specific command (`pacman -Syu`) which updates all packages and isn’t anywhere in this guide.

There is also no way anything in this guide would stop you from reimagining your deck from a recovery drive. It doesn’t touch the EFI firmware which is all you need to boot from USB. 

If you’ve really screwed up your deck, then [follow these instructions from Valve](https://help.steampowered.com/en/faqs/view/1B71-EDF2-EB6D-2BB3). I strongly recommend using a fast external NVMe or HDD instead of a USB key, as most usb sticks are painfully slow for such use (think multiple hours to reimage a deck vs less than 30 minutes). Your issue with it not booting from a recovery usb is likely impatience. When using a slow USB stick it can take 20-30 minutes or more just to boot.

**Comment by correii** [comment]
Literally all I did was run the commands in this guide to install rwfus. When I tried to run the following commands, I was getting permission errors, so I restarted my deck because that fixes 99% of all issues. When it was booting back in, it got stuck at the boot screen forever. Idk what you want me to tell you. I just followed your guide, literally only the first step, copy and pasting the commands, and it bricked my system. You don’t have to get so defensive about it. I finally managed to recover from the bricked state without losing any data, if that makes you feel any better.On Jul 22, 2024, at 6:58 AM, gbdrbob ***@***.***> wrote:﻿
Dude,  I don’t know what you did but the jank/brick warning is about not running a specific command (pacman -Syu) which updates all packages and isn’t anywhere in this guide.
There is also no way anything in this guide would stop you from reimagining your deck from a recovery drive. It doesn’t touch the EFI firmware which is all you need to boot from USB.
If you’ve really screwed up your deck, then follow these instructions from Valve. I recommend using a fast external NVMe or HDD instead of a USB key, as most usb sticks are painfully slow for such use (think multiple hours to reimage a deck vs less than 30 minutes)

**Reply by gbdrbob** [reply]
I don’t know what to tell you either. I didn’t write this guide, I’m just another person who carefully followed it and had no issues. The fact this guide was posted 18 months ago, and you’re the first person to complain about it causing an issue with your deck, should give you pause to think back carefully about what went wrong before proclaiming that the instructions are to blame and that others will inevitably have the same problem. 

If you do think you’ve found a bug in rwfus (e.g due to a problematic interaction with some other modification you’ve installed on your deck) then the place to report it is the [rwfus issue tracker](https://github.com/ValShaped/rwfus/issues). The developer and others there are better placed to try and help you establish what went wrong.

**Comment by correii** [comment]
Also no, reimaging wouldn’t work. The deck refused to boot into the boot drive selection menu while a recovery drive was inserted. I tried two different usb sticks with two different adapters. There was another menu I didn’t know about before that you can access by holding the (•••) button while booting, and that let me load from the previous SteamOS install. Which thankfully reverted whatever changes were made by the commands in the guide, while keeping all of my data intact.On Jul 31, 2024, at 6:26 AM, Calvin Corrieri ***@***.***> wrote:﻿Literally all I did was run the commands in this guide to install rwfus. When I tried to run the following commands, I was getting permission errors, so I restarted my deck because that fixes 99% of all issues. When it was booting back in, it got stuck at the boot screen forever. Idk what you want me to tell you. I just followed your guide, literally only the first step, copy and pasting the commands, and it bricked my system. You don’t have to get so defensive about it. I finally managed to recover from the bricked state without losing any data, if that makes you feel any better.On Jul 22, 2024, at 6:58 AM, gbdrbob ***@***.***> wrote:﻿
Dude,  I don’t know what you did but the jank/brick warning is about not running a specific command (pacman -Syu) which updates all packages and isn’t anywhere in this guide.
There is also no way anything in this guide would stop you from reimagining your deck from a recovery drive. It doesn’t touch the EFI firmware which is all you need to boot from USB.
If you’ve really screwed up your deck, then follow these instructions from Valve. I recommend using a fast external NVMe or HDD instead of a USB key, as most usb sticks are painfully slow for such use (think multiple hours to reimage a deck vs less than 30 minutes)

**Reply by gbdrbob** [reply]
> Also no, reimaging wouldn’t work. The deck refused to boot into the boot drive selection menu while a recovery drive was inserted. I tried two different usb sticks with two different adapters.

This is most likely a [completely separate issue](https://github.com/ValveSoftware/SteamOS/issues/1200). Try booting into the bios menu (vol + while booting) and change the USB mode from DRD to XHCI. Note that the bug report I link above states that changing the mode doesn’t help on some bios versions.

I’ve also seen [reports](https://www.reddit.com/r/SteamDeck/comments/vc6bmc/usb_flash_drive_not_detected_in_boot_manager/) of usb-a to usb-c adapters not working. Trying again with a dedicated usb-c device may help.

**Comment by TabithaSchultz** [comment]
So, I'm pretty new to linux in general. I got this to work flawlessly the first time I set it up using this guide. I had to switch all my devices to a different zt network though bc of connectivity issues my android device had with the old one. On my steam deck though, any zerotier commands were not recognized. I ended up manually deleting anything zerotier from /var/lib and /bin. I don't think I touched /etc or /usr iirc. I re-installed zerotier, didn't do anything with rwfus at first bc it was already installed. I had some trouble with connectivity, tried altering the local.conf file to see if the port was the issue, but it wasn't. Deleted /home/deck/rwfus and re-installed it. Eventually, i redid allthe steps in the guide until I'm at the point where it can hold a steady connection and can be pinged and zerotier-cli shows online and the web interface sees it online. As long as I don't reboot the system that is. If I reboot the system, zerotier-cli shows it is offline. I have to systemctl restart zerotier-one.service in order to get it back online every time I reboot. No amount of disabling/stopping it or enabling/ starting it, with our without manually doing ./rwfus before or after (idk if that would do anything, I'm just throwing spaghetti at the wall), nothing gets it to be online automatically upon boot. Idk what file I should be looking at fixing or if/how I should do a complete re-install of zerotier or rwfus. Could you please help me?

**Reply by TabithaSchultz** [reply]
Would like to add that zerotier-one.service is definitely in etc/systemd/system

**Comment by aaron777collins** [comment]
Hey there, not sure what’s going on without logs. Try getting the service logs. Another option is you can uninstall or disable this service and give my installer<https://www.reddit.com/r/SteamDeck/comments/1dllbbl/install_zerotierone_static_a_frontend_gui_flatpak/?rdt=60191> a try. It works great and shouldn’t break between any updates, etc. It also has a gui.

Sincerely,
Aaron Collins
________________________________

**Reply by TabithaSchultz** [reply]
Hi there! I have no idea how things got to where they were, and the logs (if what I found in /var/log/journal actually were the logs) made no sense. But went and tried your guide, and everything works fantastic now. systemd zerotier file runs on boot and reboot, and the GUI is really, really nice to have, thank you so much!! A note on the instructions, when cloning the repository, the stated url "https://github.com/aaron777collins/zerotier-gui.git" brings a "repository does not exist," and ended up using the url in the browser. Minor thing, and I was still able to figure it out relatively quickly for a noob, but just thought you'd like to know. Thank you again so much, this helped me so much <3

**Comment by aaron777collins** [comment]
Oh sorry I forgot to update that! My main focus was on the download script for a while and I forgot I had the other method haha. You’ll want to only use the downloader script from here on since that’s the only method I focus on now. Have fun!
________________________________

**Comment by aaron777collins** [comment]
Also you’re welcome! I’m glad you like it :)

Sincerely,
Aaron Collins
________________________________
